File: library/src/test/java/io/fotoapparat/parameter/selector/FocusModeSelectorsTest.java
package io.fotoapparat.parameter.selector;

import org.junit.Test;

import java.util.Set;

import io.fotoapparat.parameter.FocusMode;

import static io.fotoapparat.test.TestUtils.asSet;
import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;

public class FocusModeSelectorsTest {

    @Test
    public void focusMode_Available() throws Exception {
        // Given
        Set<FocusMode> availableModes = asSet(
                FocusMode.AUTO,
                FocusMode.CONTINUOUS_FOCUS,
                FocusMode.FIXED
        );

        // When
        FocusMode result = FocusModeSelectors
                .continuousFocus()
                .select(availableModes);

        // Then
        assertEquals(
                FocusMode.CONTINUOUS_FOCUS,
                result
        );
    }

    @Test
    public void focusMode_NotAvailable() throws Exception {
        // Given
        Set<FocusMode> availableModes = asSet(
                FocusMode.AUTO,
                FocusMode.FIXED
        );

        // When
        FocusMode result = FocusModeSelectors
                .continuousFocus()
                .select(availableModes);

        // Then
        assertNull(result);
    }

}

File: fotoapparat/src/test/java/io/fotoapparat/selector/FocusModeSelectorsTest.kt
package io.fotoapparat.selector

import io.fotoapparat.parameter.FocusMode
import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class FocusModeSelectorsTest {

    @Test
    fun `Select continuous focus which is available`() {
        // Given
        val availableModes = listOf(
                FocusMode.Auto,
                FocusMode.ContinuousFocusPicture,
                FocusMode.Fixed
        )

        // When
        val result = continuousFocusPicture()(availableModes)

        // Then
        assertEquals(
                expected = FocusMode.ContinuousFocusPicture,
                actual = result
        )
    }

    @Test
    fun `Select continuous focus which is not available`() {
        // Given
        val availableModes = listOf(
                FocusMode.Auto,
                FocusMode.Fixed
        )

        // When
        val result = continuousFocusPicture()(availableModes)

        // Then
        assertNull(result)
    }

}